src/krylov/pcg.c

```diff
@@ -540,18 +585,59 @@ hypre_PCGSolve( void *pcg_vdata,
          break;
       }
       alpha = gamma / sdotp;
-      if (! (alpha > HYPRE_REAL_MIN) )
+      /* alpha should always be greater zero for spd A, spd precond. and nonzero p, r*/
+      /* alpha = <r, Cr> / <p, Ap> */
+      /* for alpha close to 0 x and r will not change much unless x and/or r are close to 0 */
+      if (alpha <=0.0)
+      {
+         if (print_level > 1 && my_id == 0)
+	 {
+            hypre_printf("alpha %e", alpha);
+	 }
+         hypre_error_w_msg(HYPRE_ERROR_CONV, "Negative or zero alpha value in PCG");
+         if (skip_break == 0)
+	 {
+	    if (i == 1) { i_prod = i_prod_0; }
+            break;
+	 }
+      }
+      else if (! (alpha  >= HYPRE_REAL_TRUE_MIN))
+      {
+         hypre_error_w_msg(HYPRE_ERROR_CONV, "alpha value less than TRUE_MIN in PCG");
+         if (print_level > 1 && my_id == 0)
+	 {
+            hypre_printf("alpha %e", alpha);
+	 }
+         if (skip_break < 2)
```

Comment:
This should break for values `0` or `2` (not `1`).  This block can only happen if
(0 < alpha < HYPRE_REAL_TRUE_MIN)